### PR TITLE
Fix all the jest config

### PR DESCRIPTION
**Pull request by eapache**
- Consistently use `./` in our package script definitions.
- Don't bother specifying `testFileExtensions` to its default value.
- Set `testPathDirs`, not `testPathIgnorePatterns`. If you happen to store all
  your git checkouts in `~/src/` then the latter has unfortunate side-effects.
- Don't require a globally-installed jest in `script/test`.

@boourns cc @lemonmade

### CODE REVIEW

File: script/test [hunk 1]
```diff
@@ -2,5 +2,4 @@
 set -e -x
```
Comment (lemonmade) [inline script/test]: Does anyone know what these do? Can we just move all of this to npm scripts?

**Comment by lemonmade** [comment]
LGTM :+1:

File: script/test (hunk 1 above)
Comment (eapache) [inline script/test]: These do two things:
- make the script print the commands it is running (just a convenience)
- make the script fail when any of the commands exit non-0 (necessary for use in travis)

**Comment by boourns** [comment]
:+1:

**Pull request merged**